File: src/main/java/fi/quanfoxes/types/Short.java
package fi.quanfoxes.types;

import fi.quanfoxes.lexer.NumberType;

public class Short extends Number {
    private static final int BYTES = 2;

    public Short() {
        super(NumberType.INT16, 16, "short");
    }

    @Override
    public int getSize() {
        return BYTES;
    }
}

File: Zigzag/Types/Short.cs
public class Short : Number
{
	private const int BYTES = 2;

	public Short() : base(NumberType.INT16, 16, "short") { }

	public override int GetReferenceSize()
	{
		return BYTES;
	}

	public override int GetContentSize()
	{
		return BYTES;
	}
}
